### PR TITLE
Speed up validator build by avoiding is_file()

### PR DESCRIPTION
**Pull request by paul-m**
## Description

Adds `fromYamlRaw()` and `fromJsonRaw()` methods to `ValidatorBuilderInterface`.

## Motivation and context

Our project required testing output of hundreds of permutations of API requests. We noticed that our tests seemed rather slow, so we did some digging and found that `ValidatorBuilder` calls `is_file()` for all builds.

Adding a `fromYamlRaw()` method will allow us to bypass this call and improve performance.

If it fixes an open issue, please link to the issue here (if you write `fixes #num`
or `closes #num`, the issue will be automatically closed when the pull is accepted.)

## How has this been tested?

Firstly the PR adds test cases to `ValidatorBuilderTest` to cover the new lines of code.

Secondly, this PR adds a test called `ValidatorBuilderSpeedTest`. All it does is build a validator using whichever method is uncommented. It's not a super-great test. :-)

We also add a Composer script that runs this test 10,000 times.

Comparison results look like this:

Using `fromYaml()` and `fromJson()`:
```
Time: 00:22.492, Memory: 21.15 MB

OK (20000 tests, 20000 assertions)
```

Using `fromYamlRaw()` and `fromJsonRaw()`:
```
Time: 00:14.518, Memory: 21.15 MB

OK (20000 tests, 20000 assertions)
```

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply.

Please, please, please, don't send your pull request until all of the boxes are ticked. Once your pull request is created, it will trigger a build on our [continuous integration](http://www.phptherightway.com/#continuous-integration) server to make sure your [tests and code style pass](https://help.github.com/articles/about-required-status-checks/).

- [x] I have read the **[CONTRIBUTING](CONTRIBUTING.md)** document.
- [x] My pull request addresses exactly one patch/feature.
- [x] I have created a branch for this patch/feature.
- [x] Each individual commit in the pull request is meaningful.
- [x] I have added tests to cover my changes.
- [x] If my change requires a change to the documentation, I have updated it accordingly.

If you're unsure about any of these, don't hesitate to ask. We're here to help!

### CODE REVIEW

**Review by osteel — COMMENTED** [review]
Hey Paul! Thank you for your pull request. I understand where you come from with this and any performance gain is nice, although in this case I'd say it's fairly small, all things considered. Does it make a huge difference in your specific use case?

The "issue" I've got with this at the moment is that if we introduce public methods to load string definitions directly, I feel like there should also be methods to load files straight away.

So I'd add `fromJsonFile` and `fromYamlFile` as well.

More coherent this way I think, and we'd still not be introducing any breaking changes.

What do you think?

This would also imply restructuring the README a little bit, but you can leave that to me (I already wanted to introduce some cosmetic changes anyway, so could do that at the same time)

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# OpenAPI HttpFoundation Testing
+    # OpenAPI HttpFoundation Testing
```
[the remark this answers is not in the document]
Comment (paul-m) [reply README.md]: I think this was just me hitting tab by accident and not noticing.

File: composer.json [hunk 2]
```diff
@@ -47,6 +47,7 @@
     },
     "scripts": {
         "test": "phpunit",
+        "test-speed": "phpunit --filter ValidatorBuilderSpeedTest --repeat 10000",
```
[the remark this answers is not in the document]
Comment (paul-m) [reply composer.json]: gone. :-)

File: tests/ValidatorBuilderSpeedTest.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Osteel\OpenApi\Testing\Tests;
+
+use Osteel\OpenApi\Testing\Validator;
+use Osteel\OpenApi\Testing\ValidatorBuilder;
+
+class ValidatorBuilderSpeedTest extends TestCase
```
[the remark this answers is not in the document]
Comment (paul-m) [reply tests/ValidatorBuilderSpeedTest.php]: gone.

File: src/ValidatorBuilder.php [hunk 4]
```diff
@@ -61,6 +61,28 @@ public static function fromJson(string $definition): ValidatorBuilderInterface
         return static::fromMethod($method, $definition);
     }
 
+    /**
+     * {@inheritdoc}
+     *
+     * @param  string $definition The OpenAPI definition as YAML text.
+     * @return ValidatorBuilderInterface
+     */
+    public static function fromYamlRaw(string $definition): ValidatorBuilderInterface
```
[the remark this answers is not in the document]
Comment (paul-m) [reply src/ValidatorBuilder.php]: Renamed both, added from*File as well.

**Review by osteel — APPROVED** [review]

**Comment by osteel** [comment]
Thank you!

**Pull request merged**